### PR TITLE
Implemented emotiontime related achievements

### PR DESCRIPTION
**Pull request by sleepy-day**
PR relating to #33, let me know if there's any issues. 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new property, `Emotion`, to enhance skill representations with emotional attributes.
	- Added functionality to dynamically track and update player emotions based on their state.
	- Expanded condition checks to include emotional timing, improving gameplay interactions.

- **Bug Fixes**
	- Renamed variable for clarity, ensuring accurate representation of skill IDs related to emotions and mounts.

These updates aim to enrich user experience by adding emotional depth to skills and improving state management in gameplay.

### CODE REVIEW

File: Maple2.Server.Game/Model/Field/Actor/FieldPlayer.cs [hunk 1]
```diff
@@ -236,6 +236,15 @@ bool UpdateStateSyncTimeTracking() {
                     Session.ConditionUpdate(ConditionType.riding, codeLong: Value.Character.MapId);
                 }
                 break;
+            case ActorState.EmotionIdle:
+                if (UpdateStateSyncTimeTracking()) {
+                    Field.SkillMetadata.TryGet(stateSync.Flag1EmotionId, 1, out var emote);
+                    if (emote == null) {
+                        break;
+                    }
+                    Session.ConditionUpdate(ConditionType.emotiontime, codeString: emote.Property.Emotion);
```
Comment (Zintixx) [inline Maple2.Server.Game/Model/Field/Actor/FieldPlayer.cs:245]: You'll also need to implement target. It looks like for `emotiontime` this is the player's Map ID

File: Maple2.Server.Game/Util/ConditionUtil.cs [hunk 2]
```diff
@@ -13,6 +13,11 @@ public static bool Check(this ConditionMetadata condition, GameSession session,
 
     private static bool CheckCode(this ConditionMetadata.Parameters code, GameSession session, ConditionType conditionType, string stringValue = "", long longValue = 0) {
         switch (conditionType) {
+            case ConditionType.emotiontime:
+                if (code.Strings != null && code.Strings.Contains(stringValue)) {
```
Comment (Zintixx) [inline Maple2.Server.Game/Util/ConditionUtil.cs:17]: you can remove this. it uses the same if statement below.

**Review by Zintixx — CHANGES_REQUESTED** [review]
Nice job!

File: Maple2.Model/Game/Sync/StateSync.cs [hunk 3]
```diff
@@ -36,8 +36,8 @@ public enum Flag : byte {
     public int SyncNumber;
 
     #region Flag1
-    public int Flag1Unknown1;
-    public short Flag1Unknown2;
+    public int Flag1EmotionId; // Skill ID for emotes + flying mounts (not ground)
```
Comment (AngeloTadeucci) [inline Maple2.Model/Game/Sync/StateSync.cs:39]: Should we rename just to EmotionId?

I guess "Flag1" was used when we didn't knew what was about it

File: Maple2.Server.Game/Util/ConditionUtil.cs [hunk 4]
```diff
@@ -13,6 +13,11 @@ public static bool Check(this ConditionMetadata condition, GameSession session,
...
+                    return true;
+                }
+                break;
```
Comment (AngeloTadeucci) [inline Maple2.Server.Game/Util/ConditionUtil.cs:20]: Just add a new case, no need to repeat the check
```suggestion
            case ConditionType.emotiontime:
```

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

File: Maple2.Model/Game/Sync/StateSync.cs (hunk 3 above)
Comment (sleepy-day) [reply Maple2.Model/Game/Sync/StateSync.cs:39]: I kept the Flag1 in there as some of the other variables were named similarly like "Flag4Animation" and "Flag6Position" but I can change it if we're going with dropping the flag part

Comment (AngeloTadeucci) [reply Maple2.Model/Game/Sync/StateSync.cs]: @Zintixx should we leave "Flag1" or nah?

Comment (Zintixx) [reply Maple2.Model/Game/Sync/StateSync.cs]: I think for now it's fine. We still don't _really_ know what this entails and will need more analyzing

**Review by Zintixx — APPROVED** [review]

**Review by AngeloTadeucci — APPROVED** [review]

**Pull request merged**